Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
+    /// </summary>
+    public class EmailAddressLocalPart : IEquatable<EmailAddressLocalPart>
```

Comment:
You're probably patterning the use of `IEquatable<T>` on older parts of AutoFixture where, I must admit, I hadn't fully researched the purpose of this interface, or the contract surrounding it.

As far as I've been able to determine so far, this interface is only used in certain generic collections in the BCL, and the main purpose is to prevent boxing when comparing value types. Thus, as [the documentation](https://msdn.microsoft.com/en-us/library/ms131187.aspx) states:

*"For a value type, you should always implement IEquatable<T>"*

However, for reference types like this one, it's a lot less useful, although there may be a microscopic performance improvement to be gained. However, the question is whether it warrants the extra complexity.

Do note that the documentation furthermore states:

- *"you should also override the base class implementations of Object.Equals(Object) and GetHashCode"*
- *"you should overload the op_Equality and op_Inequality operators"*
- *"you should also implement IComparable<T> if instances of your type can be ordered or sorted"*

Given all that, I'd be more inclined to *not* implement this interface at all. `EmailAddressLocalPart` can be a great Value Object without it.

AutoFixture's legacy use of `IEnumerable<T>` is all my fault, but once one realises one's errors, one should attempt to rectify them :flushed: